src/server/svr_func.c

```diff
@@ -5102,6 +5103,28 @@ check_and_run_jobs(struct prov_vnode_info * prov_vnode_info)
 
 	rc = is_runnable(pjob, prov_vnode_info);
 	if (rc == 0) {
+		if (do_sync_mom_hookfiles || sync_mom_hookfiles_proc_running) {
+
+			/**
+			 * If mom hook files sync is in process then create
+			 * a time task where you perform this check again,
+			 * and start the job once it is done
+			 */
+
+			DBPRT(("%s: setting the time task as sync mom"
+				"hookfiles is not completed\n"
+				, __func__))
+
+			/* set a work task to run after 5 sec from now */ 
+			if (!set_task(WORK_Timed, time_now + 5, 
+				is_hooks_sync_done, pjob)){
+				log_err(errno, __func__,
+					"Unable to set task for is_hooks_sync_done; requeuing the job");
+				(void)force_reque(pjob);
+			}
+			return;
```

Comment:
> check_and_run_jobs`` is called from is_vnode_prov_done function. Before this change check_and_run_jobs was definitive in the sense that it use to either run the job or mark it as failed. Now this will make job wait until hooks are synced.

In check_and_run_job  their is a third scenario where is_runnable returns -1( i.e some nodes of the job still provisioning), then it doesn't do anything(i.e neither run and nor fail the job/provisioning). Deleting the provisioning records seems resonable since, waiting for hook_sync  is done only after provisioning finishes successfully and INUSE_PROV state is unset.